Branch-SDK/src/main/java/io/branch/referral/PrefHelper.java

```diff
@@ -255,6 +256,26 @@ public String getAPIBaseUrl() {
         }
     }
 
+    /**
+    * Overloaded version of the getAPIBaseUrl() function that allows specifying if the custom URL should be used
+     * Important for endpoints that do not have a protected version, such as the App Settings endpoint used by the Integration Validator
+    */
+    public String getAPIBaseUrl(boolean useCustom) {
+        if (useCustom && URLUtil.isHttpsUrl(customServerURL_)) {
+            return customServerURL_;
+        }
+
+        if (useEUEndpoint_) {
```

Comment:
I think you can call `getAPIBaseUrl` after the first `if`
```
       if (useCustom && URLUtil.isHttpsUrl(customServerURL_)) {
             return customServerURL_;
        }
       else {
             return getAPIBaseUrl();
       }
```

to reduce duplicate code, this make sense?

Reply:
Ah no actually because `getAPIBaseUrl` will return the custom url. Let's leave this as you've done it here.